### PR TITLE
ui: Add optional memory max to task details ribbon

### PR DESCRIPTION
**Pull request by backspace**
This is the first step in #10268.

### CODE REVIEW

File: ui/app/templates/jobs/job/task-group.hbs [hunk 1]
```diff
@@ -31,7 +31,13 @@
 
       <span class="pair" data-test-task-group-tasks><span class="term"># Tasks</span> {{this.model.tasks.length}}</span>
       <span class="pair" data-test-task-group-cpu><span class="term">Reserved CPU</span> {{format-scheduled-hertz this.model.reservedCPU}}</span>
-      <span class="pair" data-test-task-group-mem><span class="term">Reserved Memory</span> {{format-scheduled-bytes this.model.reservedMemory start="MiB"}}</span>
+      <span class="pair" data-test-task-group-mem>
+        <span class="term">Reserved Memory</span>
+        {{format-scheduled-bytes this.model.reservedMemory start="MiB"}}
+        {{#if this.model.reservedMemoryMax}}
+          ({{format-scheduled-bytes this.model.reservedMemoryMax start="MiB"}})
```
Comment (DingoEatingFuzz) [inline ui/app/templates/jobs/job/task-group.hbs:38]:
```suggestion
          ({{format-scheduled-bytes this.model.reservedMemoryMax start="MiB"}} Max)
```

File: ui/tests/acceptance/task-group-detail-test.js [hunk 2]
```diff
@@ -92,9 +93,16 @@ module('Acceptance | task group detail', function(hooks) {
       `Reserved CPU ${formatScheduledHertz(totalCPU, 'MHz')}`,
       'Aggregated CPU reservation for all tasks'
     );
+
+    let totalMemoryMaxAddendum = '';
+
+    if (totalMemoryMax) {
+      totalMemoryMaxAddendum = ` (${formatScheduledBytes(totalMemoryMax, 'MiB')})`;
```
Comment (DingoEatingFuzz) [inline ui/tests/acceptance/task-group-detail-test.js:100]:
```suggestion
      totalMemoryMaxAddendum = ` (${formatScheduledBytes(totalMemoryMax, 'MiB')} Max)`;
```

**Review by DingoEatingFuzz — CHANGES_REQUESTED** [review]
As is, this shows two numbers without an explanation of the difference. I added `Max` as suggestions.

**Comment by backspace** [comment]
Thanks for catching that oversight 😳💞

**Review by DingoEatingFuzz — APPROVED** [review]
:shipit:

**Pull request merged**